src/lightgbm/src/main/scala/LightGBMUtils.scala

```diff
@@ -71,9 +71,17 @@ object LightGBMUtils {
   def getNumExecutors(dataset: Dataset[_], defaultListenPort: Int): Int = {
     val executors = getExecutors(dataset, defaultListenPort)
     if (executors.isEmpty) {
-      // Case when run locally in local[*]
-      val blockManager = BlockManagerUtils.getBlockManager(dataset)
-      blockManager.master.getMemoryStatus.size
+      val master = dataset.sparkSession.sparkContext.master
+      // Case when run locally in local[_]
+      if (master.contains("local")) {
```

Comment:
Minor nit, I would match a regex `local[number-or-star]`. If you master happens to have a name `yarnlocalized`  or something that just happens to match `local`, you shouldn't enter this branch.

Reply:
good idea, done